systemtests/src/test/java/io/enmasse/systemtest/brokered/jms/TopicTest.java

```diff
@@ -0,0 +1,240 @@
+import org.junit.*;
+
+import javax.jms.*;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class TopicTest extends MultiTenantTestBase {
+
+    private Hashtable<Object, Object> env;
+    private Connection connection;
+    private Session session;
+    private Context context;
+    private String topic = "jmsTopic";
+    private Destination addressTopic;
+
+    @Before
+    public void setUp() throws Exception {
+        AddressSpace addressSpace = new AddressSpace("brokered-space-jms-topics", "brokered-space-jms-topics");
+        createAddressSpace(addressSpace, "none", "brokered");
+
+        addressTopic = Destination.topic(topic);
+        setAddresses(addressSpace, addressTopic);
+
+        env = new Hashtable<Object, Object>();
+        env.put(Context.INITIAL_CONTEXT_FACTORY, "org.apache.qpid.jms.jndi.JmsInitialContextFactory");
+        env.put("connectionfactory.qpidConnectionFactory", "amqps://" + getRouteEndpoint(addressSpace).toString() +
+                "?jms.clientID=testClient&" +
+                "transport.trustAll=true&" +
+                "jms.password=test&" +
+                "jms.username=test&" +
+                "transport.verifyHost=false");
+        env.put("topic." + topic, topic);
+
+        context = new InitialContext(env);
+        ConnectionFactory connectionFactory
+                = (ConnectionFactory) context.lookup("qpidConnectionFactory");
+        connection = connectionFactory.createConnection();
+        connection.start();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        deleteAddresses(addressTopic);
+        if (connection != null) {
+            connection.stop();
+        }
+        if (session != null) {
+            session.close();
+        }
+        if (connection != null) {
+            connection.close();
+        }
+    }
+
+    @Test
+    public void testMessageSubscription() throws Exception {
+        Logging.log.info("testMessageSubscription");
+        session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Topic testTopic = (Topic) context.lookup(topic);
+        MessageConsumer subscriber1 = session.createConsumer(testTopic);
+        MessageProducer messageProducer = session.createProducer(testTopic);
+
+        int count = 100;
+        List<Message> listMsgs = generateMessages(session, count);
+
+        CompletableFuture<List<Message>> received = new CompletableFuture<>();
+        Executors.newCachedThreadPool().submit(() -> {
+            received.complete(receiveMessages(subscriber1, count));
+        });
+
+        Thread.sleep(5000); //wait until subscribe
```

Comment:
I'm a bit worried that this can fail in CI. Is there a way to wait for the subscription is made before sending messages?

Reply:
I hope I found the right way for it.